src/sdk/namespace_fs.js

```diff
@@ -494,13 +494,30 @@ class NamespaceFS {
         }
     }
 
-    async _upload_stream(source_stream, upload_path, write_options) {
+    async _upload_stream(source_stream, upload_path) {
         let target_file;
+        let q_buffers = [];
+        let q_size = 0;
+        const flush = async () => {
+            if (q_buffers.length) {
+                await target_file.writev(DEFAULT_FS_CONFIG, q_buffers);
+                q_buffers = [];
+                q_size = 0;
+            }
+        };
```

Comment:
I suggest to insert this code into the try block right below the `target_file = ...` line.